src/gsy_e/models/market/forward.py

```diff
@@ -86,30 +129,48 @@ def _get_end_time(current_time: DateTime) -> DateTime:
     def _get_market_slot_duration(_current_time: DateTime, _config) -> duration:
         return duration(minutes=15)
 
+    def _calculate_closing_time(self, delivery_time: DateTime) -> DateTime:
+        """Closing time of the intraday market is 15 mins before delivery."""
```

Comment:
So currently, the closing time is the end of the former market slot, and this method is there to easily change this individually for each forward market type, right?

Reply:
Exactly, instead of reusing the duration of the market slot (which is essentially the delivery duration) we define now explicitly the closing time of the slot, and indeed, the method is overridden in order to easily change it and oblige each market type to implement it.